core/dbt/parser/base.py

```diff
@@ -396,61 +393,53 @@ def update_parsed_node_config(
 
         # unrendered_config is used to compare the original database/schema/alias
         # values and to handle 'same_config' and 'same_contents' calls
-        parsed_node.unrendered_config = config.build_config_dict(
+        parsed_node.unrendered_config = config_builder.build_config_dict(
             rendered=False, patch_config_dict=patch_config_dict
         )
 
-        parsed_node.config_call_dict = config._config_call_dict
-        parsed_node.unrendered_config_call_dict = config._unrendered_config_call_dict
+        parsed_node.config_call_dict = config_builder._config_call_dict
+        parsed_node.unrendered_config_call_dict = config_builder._unrendered_config_call_dict
 
         # do this once before we parse the node database/schema/alias, so
         # parsed_node.config is what it would be if they did nothing
-        self.update_parsed_node_config_dict(parsed_node, config_dict)
+        self.clean_and_fix_config_dict(parsed_node, config_dict)
         # This updates the node database/schema/alias/relation_name
         self.update_parsed_node_relation_names(parsed_node, config_dict)
 
         # tests don't have hooks
         if parsed_node.resource_type == NodeType.Test:
             return
 
-        # at this point, we've collected our hooks. Use the node context to
-        # render each hook and collect refs/sources
+        # Use the node context to render each hook and collect refs/sources.
         assert hasattr(parsed_node.config, "pre_hook") and hasattr(parsed_node.config, "post_hook")
         hooks = list(itertools.chain(parsed_node.config.pre_hook, parsed_node.config.post_hook))
         # skip context rebuilding if there aren't any hooks
         if not hooks:
             return
         if not context:
-            context = self._context_for(parsed_node, config)
+            context = self._context_for(parsed_node, config_builder)
         for hook in hooks:
             get_rendered(hook.sql, context, parsed_node, capture_macros=True)
 
-    def initial_config(self, fqn: List[str]) -> ContextConfig:
-        config_version = min([self.project.config_version, self.root_project.config_version])
-        if config_version == 2:
```

Comment:
Is it safe to remove this check entirely at this point?

Reply:
The config_version == 2?  Yeah, we haven't required that or checked in at least a year and a half. I think pre-version-2 was way back in... version 14?  Before I started anyway.